File: 167-two-sum-II-input-array-Is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers) - 1
        while l<r:
            if numbers[l] > 0 and target > 0:
                while numbers[r] > target:
                    r -= 1
            if numbers[r] < 0 and target < 0:
                while numbers[l] < target:
                    l += 1
            if numbers[l] + numbers[r] == target:
                return [l+1, r+1]
            elif numbers[l] + numbers[r] > target:
                r -= 1
            elif numbers[l] + numbers[r] < target:
                l += 1
        return []

File: cpp/167-two-sum-II-input-array-Is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        vector<int> ret_indices;
        int l=0, r=numbers.size()-1;
        while (l<r){
            if(numbers[l]+numbers[r] == target){
                ret_indices.push_back(l+1);
                ret_indices.push_back(r+1);
                return ret_indices;
            }
            else if (numbers[l]+numbers[r] > target){
                r--;
            }
            else{
                l++;
            }
        }
        return ret_indices;
    }
};